### ISSUE TITLE
Lint/NumberConversion autocorrect breaks nil-safe behavior of safe navigation

### ISSUE THREAD

**Issue by RedZapdos123** [rubocop/rubocop#15252]
## Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

## Issue type

Incorrect autocorrect

## Cop name

`Lint/NumberConversion`

## Steps to reproduce the problem

I reproduced this on the latest stable RuboCop release and also on current `master`.

1. Create a file `example.rb` with the following content:

```ruby
foo = nil
p foo&.to_i
```

2. Run:

```bash
bundle exec ruby exe/rubocop example.rb --only Lint/NumberConversion -A
ruby example.rb
```

After autocorrect, RuboCop rewrites the file to:

```ruby
foo = nil
p Integer(foo, 10)
```

## Expected behavior

Autocorrect should preserve the nil-safe behavior of safe navigation, or skip autocorrection if that cannot be done safely.

In this example, `foo&.to_i` returns `nil` when `foo` is `nil`, so autocorrect should not change it into code that raises.

## Actual behavior

`Lint/NumberConversion` registers an offense for `foo&.to_i` and autocorrects it to `Integer(foo, 10)`.

That changes runtime behavior:

- before autocorrect: `p foo&.to_i` prints `nil`
- after autocorrect: `Integer(foo, 10)` raises `ArgumentError: base specified for non string value`

This seems like incomplete handling of safe navigation for variable receivers. The current autocorrect preserves the syntax shape, but it does not preserve the semantics of `&.` when the receiver can be `nil`.

## RuboCop configuration

```yaml
# No special configuration
```

## RuboCop version

```shell
1.87.0 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 4.0.2) +Rubydex [x86_64-linux]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.10.1
```

**Comment by RedZapdos123** [comment]
Opened a PR to resolve this issue.
